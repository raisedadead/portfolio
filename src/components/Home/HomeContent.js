import React from 'react';
import { ClassNames } from '@emotion/core';

import { StaticQuery, graphql } from 'gatsby';
import Box from 'mineral-ui/Box';
import Text from 'mineral-ui/Text';
import Flex from 'mineral-ui/Flex';
import Tooltip from 'mineral-ui/Tooltip';
import Img from 'gatsby-image';

import LinkItem from '../Link';

const Link = props => (
  <ClassNames>
    {({ css }) => (
      <LinkItem
        className={css`
          outline: none;
          text-decoration: none;
          padding: 2px 1px 0;
          border-bottom-style: dashed;
          border-bottom-color: currentcolor;
          border-bottom-width: 1px;
          color: currentcolor;
        `}
        {...props}
      />
    )}
  </ClassNames>
);

export const PureHomeContent = ({ data }) => (
  <Box
    as="section"
    marginHorizontal="auto"
    breakpoints={[`narrow`, `medium`, `wide`]}
    width={[`80vw`, `80vw`, `95vw`, `50vw`]}
  >
    <Flex
      alignItems="center"
      justifyContent="center"
      direction="column"
      height="75vh"
    >
      <Box
        breakpoints={[`narrow`, `medium`, `wide`]}
        width={[`25vw`, `25vw`, `15vw`, `15vw`]}
        margin="1em"
      >
        <Img sizes={data.profileImage.childImageSharp.fluid} />
      </Box>
      <Text as="h2" align="center">
        <Tooltip cursor="help" content="/mru:dʒi:eʃ/" placement="top" usePortal>
          mrugesh
        </Tooltip>
        {` `}
        mohapatra
      </Text>

      <Text as="h3" align="center">
        developer. music addict. open source enthusiast. noob photographer.
      </Text>
      <br />
      <Text as="p" align="center">
        Lead Developer and Community Advocate at{`  `}
        <Link to="https://www.freecodecamp.org" aria-label="freecodecamp.org">
          freeCodeCamp.org
        </Link>
      </Text>
    </Flex>
  </Box>
);

export const HomeContent = () => (
  <StaticQuery
    query={graphql`
      query {
        profileImage: file(relativePath: { eq: "profile.png" }) {
          childImageSharp {
            fluid(maxWidth: 1000) {
              ...GatsbyImageSharpFluid_withWebp_tracedSVG
            }
          }
        }
      }
    `}
    render={data => <PureHomeContent data={data} />}
  />
);

export default HomeContent;
