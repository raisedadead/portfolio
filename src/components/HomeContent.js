import React from 'react';
import { StaticQuery, graphql } from 'gatsby';
import Box from 'mineral-ui/Box';
import Text from 'mineral-ui/Text';
import Flex from 'mineral-ui/Flex';
import Img from 'gatsby-image';

export const PureHomeContent = ({ data }) => (
  <Box
    as="section"
    marginHorizontal="auto"
    breakpoints={[`narrow`, `medium`, `wide`]}
    width={[`80vw`, `80vw`, `95vw`, `50vw`]}
  >
    <Flex
      alignItems="center"
      justifyContent="center"
      direction="column"
      height="75vh"
    >
      <Box
        breakpoints={[`narrow`, `medium`, `wide`]}
        width={[`25vw`, `25vw`, `15vw`, `15vw`]}
        margin="1em"
      >
        <Img sizes={data.profileImage.childImageSharp.fluid} />
      </Box>
      <Text as="h1" align="center">
        mrugesh mohapatra
      </Text>
      <Text as="h2" align="center">
        developer. music addict. open source enthusiast. noob photographer.
      </Text>
      <Text as="h4" align="center">
        Developer Advocate at freeCodeCamp.org
      </Text>
    </Flex>
  </Box>
);

export const HomeContent = () => (
  <StaticQuery
    query={graphql`
      query {
        profileImage: file(relativePath: { eq: "profile.png" }) {
          childImageSharp {
            fluid(maxWidth: 1000) {
              ...GatsbyImageSharpFluid_withWebp_tracedSVG
            }
          }
        }
      }
    `}
    render={data => <PureHomeContent data={data} />}
  />
);

export default HomeContent;
