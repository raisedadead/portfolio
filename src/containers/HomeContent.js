import React from 'react'
import { StylesHelper } from '../components/'

const heading = StylesHelper('uk-heading-primary')
const tagline = StylesHelper('uk-text-large')
const content = StylesHelper('uk-width-1-3')
export default props => (
  <div {...props}>
    <img src="assets/images/profile.png" alt="profile" className={content} />
    <p
      className={heading}
      style={{ color: '#F5F5F5', fontFamily: "'Lora', sans-serif" }}
    >
      mrugesh mohapatra
    </p>
    <p
      className={tagline}
      style={{ color: '#D5F7FF', fontFamily: "'Montserrat', sans-serif" }}
    >
      developer. music addict. open source enthusiast. noob photographer.
    </p>
  </div>
)
