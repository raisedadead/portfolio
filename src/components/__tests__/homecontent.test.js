import React from 'react';
import renderer from 'react-test-renderer';

import { PureHomeContent as HomeContent } from '../HomeContent';

describe(`HomeContent`, () => {
  it(`renders correctly`, () => {
    const data = {
      profileImage: {
        childImageSharp: {
          fluid: {
            tracedSVG: `random-2b80985b4264e4cb34b1850de91c6911`,
            aspectRatio: 1,
            src: `random-2b80985b4264e4cb34b1850de91c6911`,
            srcSet: `random-2b80985b4264e4cb34b1850de91c6911`,
            srcWebp: `random-2b80985b4264e4cb34b1850de91c6911`,
            srcSetWebp: `random-2b80985b4264e4cb34b1850de91c6911`,
            sizes: `(max-width: 300px) 100vw, 300px`
          }
        }
      }
    };
    const tree = renderer.create(<HomeContent data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
