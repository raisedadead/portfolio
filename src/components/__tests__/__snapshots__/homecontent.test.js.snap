// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContent renders correctly 1`] = `
<section
  className="css-1hm9igd elghiqq0"
>
  <div
    className="e7r6m60 css-13v9bvk elghiqq0"
  >
    <div
      className="css-1nsbwhm elghiqq0"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          src="random-2b80985b4264e4cb34b1850de91c6911"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 0.5s",
              "transitionDelay": "0.25s",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 300px) 100vw, 300px"
            srcSet="random-2b80985b4264e4cb34b1850de91c6911"
            type="image/webp"
          />
          <img
            alt=""
            onLoad={[Function]}
            sizes="(max-width: 300px) 100vw, 300px"
            src="random-2b80985b4264e4cb34b1850de91c6911"
            srcSet="random-2b80985b4264e4cb34b1850de91c6911"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source type='image/webp' srcset=\\"random-2b80985b4264e4cb34b1850de91c6911\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" /><img sizes=\\"(max-width: 300px) 100vw, 300px\\" srcset=\\"random-2b80985b4264e4cb34b1850de91c6911\\" src=\\"random-2b80985b4264e4cb34b1850de91c6911\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <h1
      className="css-pdf1eb e19le4zx0"
    >
      mrugesh mohapatra
    </h1>
    <h2
      className="css-pdf1eb e19le4zx0"
    >
      developer. music addict. open source enthusiast. noob photographer.
    </h2>
    <h4
      className="css-pdf1eb e19le4zx0"
    >
      Developer Advocate at freeCodeCamp.org
    </h4>
  </div>
</section>
`;
